src/Compilers/CSharp/Portable/FlowAnalysis/UnassignedFieldsWalker.cs

```diff
@@ -259,6 +259,11 @@ void markMemberAsAssigned(string notNullMember, ref LocalState state)
                     default:
                         continue;
                 }
+                var annotations = (field.IsImplicitlyDeclared ? field.AssociatedSymbol : field).GetFlowAnalysisAnnotations();
+                if ((annotations & FlowAnalysisAnnotations.MaybeNull) != 0)
+                {
+                    continue;
+                }
```

Comment:
Consider moving down to after `var symbol = ...` and use `symbol` directly rather than using `AssociatedSymbol` here.

Reply:
That looks like it would be cleaner, thanks